src/subtyping.md

```diff
@@ -15,5 +15,70 @@ fn bar<'a>() {
     let t: &'a str = s;
 }
 ```
-Since `'static` "lives longer" than `'a`, `&'static str` is a subtype of
-`&'a str`.
+
+Since `'static` outlives `'a`, `&'static str` is a subtype of `&'a str`.
+
+Subtyping also exists for [higher-ranked] function pointer and trait object
+types. Replacing a higher ranked lifetime with a concrete lifetime produces a
+subtype.
+
+```rust
+fn fun_ptr() {
+    // Explicitly f: &(for<'a> fn(&'a i32) -> &'a i32).
+    let f: &(fn(&i32) -> &i32) = &((|x| x) as fn(&i32) -> &i32);
+    let g: &(fn(&'static i32) -> &'static i32) = f;
+}
+
+fn fun_trait() {
+    // Explicitly f: &(for<'a> Fn(&'a i32) -> &'a i32).
+    let f: &(Fn(&i32) -> &i32) = &(|x| x);
+    let g: &(Fn(&'static i32) -> &'static i32) = f;
+}
```

Comment:
Unless I'm mistaken, doctest implicitly wraps the example in `main`, so you can get rid of these distracting functions.

Also consider renaming f and g to "base" and "subtype" or something more self-descriptive.